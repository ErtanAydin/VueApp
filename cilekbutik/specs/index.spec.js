import { mount, createLocalVue, RouterLinkStub } from "@vue/test-utils";
import index from "@/pages/index";
import Vuex from "vuex"

const localVue = createLocalVue()
localVue.use(Vuex)

jest.useFakeTimers()

const store = new Vuex.Store({
  state: {
    slides: [],
    Urunler: []
  },
  actions: {
    slidesfetch(){
      let slides = [
        {
          url:'https://www.cilekbutik.net/cdn/1/300/500/images/urunler/5fa57c217b440-1353.jpg',
        },
        {
          url:'https://www.cilekbutik.net/cdn/2/100/100/images/urunler/5fa6a282db005-1368.jpg',
        }]
        this.state.slides=slides
    },
    Urunlerfetch(){
      let Urunler = [
        {
            id:0,
            url:'https://www.cilekbutik.net/cdn/1/300/500/images/urunler/5fa57c217b440-1353.jpg',
            name:'Lakers Sweat',
            fiyat:55.00,
            url2:'https://www.cilekbutik.net/cdn/2/100/100/images/urunler/5fa6a282db005-1368.jpg',
            barkod:"6242005630101",
            kategori:"Ceket",
            bedenler:[{ad:'S'},{ad:'M'},{ad:'L'}],
            beden:'M',
            renk:'TURUNCU',
            tutar:55.00,
            adet:10,
        }]
        this.state.Urunler=Urunler
    }
  }
})

describe('init', () => {
  test('interval test', () => {
    const wrapper = mount(index, { 
      store, 
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })
    expect(wrapper.vm.interval).not.toBe(undefined)
  })

  test('visibleSlide artış test', () => {
    const wrapper = mount(index, { 
      store, 
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })
    expect(wrapper.vm.visibleSlide).toBe(0)
    jest.advanceTimersByTime(5000)
    expect(wrapper.vm.visibleSlide).toBe(1)
    jest.advanceTimersByTime(5000)
    expect(wrapper.vm.visibleSlide).toBe(0)
  })
})

describe('next', () => {
  test('next test', () => {
    const wrapper = mount(index, { 
      store, 
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })
    wrapper.vm.next()
    expect(wrapper.vm.visibleSlide).toEqual(1)
    wrapper.vm.next()
    expect(wrapper.vm.visibleSlide).toEqual(-1)
  })
})

describe('prev', () => {
  test('prev test', () => {
    const wrapper = mount(index, { 
      store, 
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })
    wrapper.vm.prev()
    expect(wrapper.vm.visibleSlide).toEqual(1)
    wrapper.vm.prev()
    expect(wrapper.vm.visibleSlide).toEqual(0)
  })
})
 
describe('mounted', () => {
  test('mounted test', () => {
    const wrapper = mount(index, { 
      store, 
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })
    expect(wrapper.vm.interval).not.toBe(undefined)
  })
})

describe('created', () => {
  test('created test', () => {
    expect(store.state.slides).not.toBe(undefined)
  })
})

describe("index", () => {
  test("mounts properly", () => {
    const wrapper = index;
    expect(wrapper).toBeTruthy();
  });

  test("gerçek Vuex Store kullanarak index render edildi", () => {
    const wrapper = mount(index, { 
      store, 
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  });
});