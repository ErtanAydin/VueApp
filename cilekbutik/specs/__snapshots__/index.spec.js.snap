// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index gerçek Vuex Store kullanarak index render edildi 1`] = `
<div>
  <div class="CarouselBody">
    <div class="carousel Carousel">
      <transition-stub name="crossfade">
        <div class="carousel-slide"><a href="#"><img src="https://www.cilekbutik.net/cdn/1/300/500/images/urunler/5fa57c217b440-1353.jpg"></a></div>
      </transition-stub>
      <transition-stub name="crossfade">
        <div class="carousel-slide" style="display: none;"><a href="#"><img src="https://www.cilekbutik.net/cdn/2/100/100/images/urunler/5fa6a282db005-1368.jpg"></a></div>
      </transition-stub> <button class="next carousel-button" style="font-size: x-large;"><i id="normal-text" class="fas fa-chevron-right"></i></button> <button class="prev carousel-button" style="font-size: x-large;"><i id="normal-text" class="fas fa-chevron-left"></i></button>
    </div>
  </div>
  <div class="UrunlerBody">
    <div class="CokSatanUrunler">
      ÇOK SATAN ÜRÜNLER
    </div>
    <div class="Urunler">
      <div class="UrunAlanı"><a><img title="Lakers Sweat" src="https://www.cilekbutik.net/cdn/1/300/500/images/urunler/5fa57c217b440-1353.jpg" alt="" class="UrunImage"></a> <a class="LINK">
          <p title="Lakers Sweat" class="UrunIsim"> Lakers Sweat</p>
        </a> <a class="LINK"><b title="Lakers Sweat" class="UrunFiyat"><span>55,00 TL </span></b></a> <button class="SepeteEkleButton"> Sepete Ekle </button></div>
    </div>
  </div>
</div>
`;
